### DISCUSSION TITLE
How to disable navigation.sections on blog?

### DISCUSSION THREAD

**Discussion by madjxatw** [squidfunk/mkdocs-material#6865] [Q&A]
I disables `navigation.sections` to get a collapsible navigation sidebar, but this leads to the `Archive` and `Categories` sections on the blog home page being collapsed as well.

![screenshot_20240304-032427Z](https://github.com/squidfunk/mkdocs-material/assets/6035833/42ae08c8-0fb9-4415-81d9-e63eff0d297b)

I want to let  `Archive` and `Categories` rendered as groups in sidebar while keeping other non-blog docs having a collapsible navigation sidebar. How to achieve this?

**Comment by kamilkrzyskow** [comment]
Hello @madjxatw,
I don't see any "override features" option on https://squidfunk.github.io/mkdocs-material/plugins/blog/ and the Jinja `base.html` template defines the features directly based on the theme config:
https://github.com/squidfunk/mkdocs-material/blob/15f15068d958506734d3544108a1e6971d7e511b/src/templates/base.html#L195

So there is no config option to get the result you want.

However, to get what you want, you can use a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) that would modify the `features` between the built pages. MkDocs builds the pages sequentially one after the other, so this approach is valid and shouldn't have mix ups.

Here is a quick example, with debug prints to see what is happening, there is no validation or safety checks for fat fingering the keyboard nor proper logging, but it's enough to get you started :v:
The [events](https://www.mkdocs.org/dev-guide/plugins/#events) run just before the page is rendered with the theme templates and just before the page is saved to disk (before another page begins to be rendered).

<details open>
<summary> Hook </summary>

`overrides/.hooks/sections_on_blog.py`

```py
from pprint import pprint as pp

class CopyPassAlong:

    context = None
    obj = None

def on_page_context(context, page, config, nav):
    
    features = context["config"]["theme"]["features"]
    
    # Skip non-blog pages
    if not page.url.startswith("blog"):
        print(f"non blog: {page.url}")
        pp(features)
        return
        
    print(f"blog - before change: {page.url}")
    pp(features)
    
    # Store the context and copy in global class to pass it along to the other event
    CopyPassAlong.context = context
    CopyPassAlong.obj = list(features)
    
    # Enable the feature on the blog page
    context["config"]["theme"]["features"].append("navigation.sections")

def on_post_page(output, page, config):
    
    # Skip non-blog pages
    if not page.url.startswith("blog"):
        return
        
    print(f"blog - after change - before restore: {page.url}")
    features = CopyPassAlong.context["config"]["theme"]["features"]
    pp(features)
    
    # Restore the features from the copy
    CopyPassAlong.context["config"]["theme"]["features"] = CopyPassAlong.obj
    
    print(f"blog - after restore: {page.url}")
    features = CopyPassAlong.context["config"]["theme"]["features"]
    pp(features)
```

</details>
